cypress/integration/5-image/image-page.spec.js

```diff
@@ -0,0 +1,519 @@
+                    cy.addImagesToUser(deletableUsername, [
+                        {
+                            data: fixtureImageData,
+                            uploadeddate_str: new Date().toString(),
+                            mimetype: "image/jpeg",
+                        },
+                    ]);
+                    cy.wait(1000);
+                    cy.getImagesForUser(deletableUsername).then((imagesRes) => {
+                        assert.isAtLeast(imagesRes.length, 1);
+                        deletableImageID = imagesRes[0].id;
+                        cy.intercept("DELETE", `/images/${deletableImageID}`).as("imageDelete");
+                        cy.intercept("GET", "/").as("index");
+                        cy.visit(`/images/${deletableImageID}`);
+                    });
+                });
+        });
+        it("should allow user to delete image if 'Delete' link is clicked", () => {
+            cy.getImagesForUser(deletableUsername).should("have.length", 1);
+
+            cy.get("#delete").should("be.visible").click();
+
+            cy.get("#delete-confirm-yesno > span:first-child").click();
+
+            cy.wait("@imageDelete").its("response.statusCode").should("eq", 200);
+
+            cy.wait("@index");
+
+            cy.getImagesForUser(deletableUsername).should("have.length", 0);
+
+            cy.request({
+                method: "GET",
+                url: `/images/${deletableImageID}`,
+                failOnStatusCode: false,
+            })
+                .its("status")
+                .should("eq", 404);
+        });
+        it("should not allow image to be deleted if user is not logged in when they confirm action", () => {
+            cy.getImagesForUser(deletableUsername).should("have.length", 1);
+
+            cy.get("#delete").should("be.visible").click();
+
+            cy.logout();
+
+            cy.get("#delete-confirm-yesno > span:first-child").click();
+
+            cy.wait("@imageDelete").its("response.statusCode").should("eq", 401);
+
+            cy.get("@index").should("be.null");
+
+            cy.getImagesForUser(deletableUsername).should("have.length", 1);
+
+            cy.request({
+                method: "GET",
+                url: `/images/${deletableImageID}`,
+                failOnStatusCode: false,
+            })
+                .its("status")
+                .should("eq", 200);
+        });
+        it("should handle error when performing delete action", () => {
+            cy.intercept("DELETE", `/images/${deletableImageID}`, {
+                statusCode: 500,
+                body: {
+                    status: "error",
+                    message: "Cannot delete image. Server error occurred.",
+                },
+            }).as("imageDelete");
+
+            cy.getImagesForUser(deletableUsername).should("have.length", 1);
+
+            cy.get("#delete").should("be.visible").click();
+
+            cy.logout();
+
```

Comment:
```suggestion
```
Not needed here since we're testing server error not authorization error. (the previous test covers that)